bigtable/bigtable-proxy/src/main/java/com/google/cloud/bigtable/examples/proxy/metrics/MetricsImpl.java

```diff
@@ -0,0 +1,227 @@
+            .ofLongs()
+            .build();
+
+    responseSizes =
+        meter
+            .histogramBuilder(METRIC_PREFIX + "client.call.rcvd_total_message_size")
+            .setDescription(
+                "Total bytes received per call from Bigtable service (excluding metadata, grpc and  transport framing bytes)")
+            .setUnit("by")
+            .ofLongs()
+            .build();
+
+    gfeLatency =
+        meter
+            .histogramBuilder(METRIC_PREFIX + "client.gfe.duration")
+            .setDescription(
+                "Latency as measured by Google load balancer from the time it "
+                    + "received the first byte of the request until it received the first byte of the "
+                    + "response from the Cloud Bigtable service.")
+            .setUnit("ms")
+            .build();
+
+    gfeResponseHeadersMissing =
+        meter
+            .counterBuilder(METRIC_PREFIX + "client.gfe.duration_missing.count")
+            .setDescription("Count of calls missing gfe response headers")
+            .setUnit("{call}")
+            .build();
+
+    clientCallLatencies =
+        meter
+            .histogramBuilder(METRIC_PREFIX + "client.call.duration")
+            .setDescription("Total duration of how long the outbound call took")
+            .setUnit("ms")
+            .build();
+
+    meter
+        .gaugeBuilder(METRIC_PREFIX + "client.call.max_outstanding_count")
+        .setDescription("Number of concurrent")
+        .setUnit("{call}")
+        .ofLongs()
+        .buildWithCallback(o -> o.record(maxSeen.getAndSet(0)));
+  }
+
+  @Override
+  public void close() {
+    meterProvider.close();
+  }
+
+  private static SdkMeterProvider createMeterProvider(Credentials credentials, String projectId)
+      throws IOException {
+    MetricServiceSettings.Builder metricServiceSettingsBuilder = MetricServiceSettings.newBuilder();
+    metricServiceSettingsBuilder
+        .setCredentialsProvider(FixedCredentialsProvider.create(credentials))
+        .setTransportChannelProvider(
+            FixedTransportChannelProvider.create(
+                GrpcTransportChannel.create(
+                    ManagedChannelBuilder.forTarget(
+                            MetricConfiguration.DEFAULT_METRIC_SERVICE_ENDPOINT)
+                        // default 8 KiB
+                        .maxInboundMetadataSize(16 * 1000)
+                        .build())))
+        .createMetricDescriptorSettings()
+        .setSimpleTimeoutNoRetriesDuration(
+            Duration.ofMillis(MetricConfiguration.DEFAULT_DEADLINE.toMillis()))
+        .build();
+
+    MetricConfiguration config =
+        MetricConfiguration.builder()
+            .setProjectId(projectId)
+            .setMetricServiceSettings(metricServiceSettingsBuilder.build())
+            .setInstrumentationLibraryLabelsEnabled(false)
+            .build();
+
+    MetricExporter exporter = GoogleCloudMetricExporter.createWithConfiguration(config);
```

Comment:
Did you test exporting the metrics? Wondering if we need to create the metrics descriptor?

Reply:
And also which monitoed resource it's exported to by default?

Reply:
I think thats out of scope....I'm using the MetricExporter for exporting metrics, so I'm relying on its correctness. The cost of doing a full end to end test is quite high (waiting for 5 minutes for publication) and the benefit is quite low.

It will use the autodetected monitored resource that GoogleCloudMetricExporter uses by default....GCE VM if on gce, gke pod if on gke, generic node if on prem, etc

Reply:
If that's the case can we have the option to not export metrics?